ext/src/content/content.js

```diff
@@ -52,89 +44,131 @@ function init() {
+    const COMMENT = "Comment";
+    const REPLY = "Reply…";
+    const CANCEL = "Cancel";
+    const CLOSE_ISSUE = " Close issue";
+    const CLOSE_PULL_REQUEST = " Close pull request";
+    const SUBMIT_PULL_REQUEST = "Create pull request";
+    const SUBMIT_NEW_ISSUE = "Submit new issue";
+    const BUTTONS_TO_SEARCH_FOR = [
+      COMMENT,
+      CANCEL,
+      CLOSE_ISSUE,
+      CLOSE_PULL_REQUEST,
+      SUBMIT_PULL_REQUEST,
+      SUBMIT_NEW_ISSUE
+    ];
+
+    const buttons = document.getElementsByTagName("button");
+    const createList = [];
+    for (const button of buttons) {
+      if (BUTTONS_TO_SEARCH_FOR.includes(button.innerText)) {
+        if (
+          button.innerText === COMMENT &&
+          (button.parentNode.parentNode.querySelector(
+            "button[name=comment_and_close]"
+          ) ||
+            button.parentNode.parentNode.querySelector(
+              "button[data-confirm-cancel-text]"
+            ))
+        ) {
+          continue;
         }
-        return;
+        createList.push(button);
       }
-
-      if (activeButtons.has(target)) {
-        return;
-      }
-
-      let parentEl;
-      let buttonsRowSelector;
-      let isNewConversationComment = false;
-
-      if (document.querySelector(DIFF_VIEW_CONTAINER_CLASS)) {
-        parentEl = getParentWithClass(target, "line-comments");
-        buttonsRowSelector = ".form-actions";
-      } else {
-        const newConversationComment = getParentWithClass(
-          target,
-          "timeline-new-comment"
+      if (button.innerText === REPLY) {
+        const observer = new MutationObserver(() => {
+          discoverButtonsAndCreatePrettierButtons();
+        });
+        observer.observe(
+          findWithClass(button, "inline-comment-form-container"),
+          { attributes: true }
         );
-        isNewConversationComment = !!newConversationComment;
-
-        if (isNewConversationComment) {
-          parentEl = newConversationComment;
-          buttonsRowSelector = `#partial-new-comment-form-actions .d-flex`;
-        } else {
-          parentEl = getParentWithClass(target, "previewable-comment-form");
-          buttonsRowSelector = ".form-actions";
-        }
       }
+    }
+    return createList;
+  }
 
-      const buttonsRowEl = parentEl.querySelector(buttonsRowSelector);
-      const style = isNewConversationComment ? { marginRight: "4px" } : {};
-      const buttonEl = renderButton(buttonsRowEl, {
-        append: !isNewConversationComment,
-        classes: ["prettier-btn"],
-        style
-      });
-      activeButtons.set(target, buttonEl);
-      buttonEl.addEventListener("click", event => {
-        event.preventDefault();
-        target.value = window.prettier.format(target.value, {
-          parser: "markdown",
-          plugins: window.prettierPlugins
+  function discoverButtonsAndCreatePrettierButtons() {
+    const BUTTON_STYLE = { float: "left", "margin-right": "10px" };
```

Comment:
I think where the button is rendering and using `float` here is causing some of the wonkiness with styles. It would be great if we could render it in the `div` containing the other buttons (the original iteration did this well, I think). Again, not a blocker, but leaving this comment for posterity and for the person who will work on this in the future.

Reply:
Agreed, the issue is mainly that a lot of different buttons have different divs they are contained in (eg. 'Close pull request' and 'Comment' buttons are actually within separate div elements) but can look into styling that gives consistent results across different sets of buttons.